### PR TITLE
Cm/flexbox vs grid

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description
A resource comparing Flexbox vs Grid.  

## Questions for the Author
- [x] Have you updated the weekly learning goals in the pedagogy resource

## Todos
Are any of these items still in progress?
- [x] Update the README with the link
- [x] Move Codepens here to the Ada Account

## Feedback Requested
Happy to make changes, it's still rough.

### CODE REVIEW

File: 05-html-css/flexbox-vs-css-grid.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+header {
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+}
+img {
+  margin: .5em;
+}
+body {
+  background-color: black;
+}
```
Comment (droberts-ada) [inline 05-html-css/flexbox-vs-css-grid.md]: You don't have this `body` ruleset above when talking about flexbox.

File: 05-html-css/flexbox-vs-css-grid.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+The content is organized into both rows and columns.  The grid generates as many 3-column rows as needed to contain the content.  We see gaps where content does not fill an entire cell, and with only 5 images, the bottom row contains an empty cell.  
+
+With Flexbox the layout responds to the content, whereas with Grid the content responds to the layout.  It is often said that _Flexbox works from the flex-items out_ whereas _Grid works from the layout in_.
+
+### Flexbox Adjusts Automatically To Screen width
+
+Because Flexbox adjusts to the content the layout will respond automatically if you narrow the browser window adjusting the number of items in each row to allow them to fit.  
+
+By contrast, Grid example above will always maintain 3 columns.  There are ways to adjust the number of columns for smaller screens, by using something like: `grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));` which will generate as many columns as needed with a minimum size of 200 pixels and a maximum of 1 fraction of the row.  
+
```
Comment (droberts-ada) [inline 05-html-css/flexbox-vs-css-grid.md]: With that repeat/autofill example, a dozen students' heads just exploded. They're going to want to know a lot more about this, or we should not show the complex example.

File: 05-html-css/flexbox-vs-css-grid.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+- Do I only need to lay content out by row **or** by column?
+  - Use Flexbox
+- Do I need to control layout with both row **and** column?
+  - Use Grid
+- Is it important that items "line up" with each other?
+  - Use Grid
+
+## Browser Support
+
+It should be said that CSS Grid is newer than Flexbox and has less support.  However as of March 12, 2018 [88% of web traffic supports Grid](https://caniuse.com/#feat=css-grid).  This means if you need to support older browsers, you will need to consider alternatives to Grid.  There are strategies to create fallback styles to handle older browsers which we will not cover here, but you can [read about them](https://rachelandrew.co.uk/css/cheatsheets/grid-fallbacks).  At Ada we will use the modern Chrome browser as our standard and Chrome fully supports Grid.
```
Comment (droberts-ada) [inline 05-html-css/flexbox-vs-css-grid.md]: I think I would not even include this section. We've talked about browser support a couple other places, I think covering it more is going to scare them off.

Comment (droberts-ada) [reply 05-html-css/flexbox-vs-css-grid.md]: You'd have to remove that bit from the conclusion as well.

File: 05-html-css/flexbox-vs-css-grid.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+## So do I use Flexbox or Grid?
+
+Ask yourself a couple of questions:
+- Do I only need to lay content out by row **or** by column?
+  - Use Flexbox
+- Do I need to control layout with both row **and** column?
+  - Use Grid
+- Is it important that items "line up" with each other?
+  - Use Grid
```
Comment (tildeee) [inline 05-html-css/flexbox-vs-css-grid.md]: what does "line up" mean?

I think flexbox does alignment better

**Pull request merged**